addons/html_editor/static/src/main/list/list_plugin.js

```diff
@@ -379,7 +379,10 @@ export class ListPlugin extends Plugin {
     }
 
     mergeSimilarLists(element) {
-        if (!element.matches("ul, ol, li.oe-nested")) {
+        if (
+            !element.matches("ul, ol") &&
+            !(element.matches("li.oe-nested") && element.querySelector("ul, ol"))
+        ) {
```

Comment:
Let's go for
```suggestion
        if (
            !element.matches("ul, ol, li.oe-nested") ||
            (element.matches("li.oe-nested") && !element.querySelector("ul, ol"))
        ) {
```
easy to understand.